### PR TITLE
operator: Skip failed operator to support batch operators in  `AddWaitingOperator`

### LINKED ISSUE #4643: Consider the situation when adding batch operators but failed in check
### Describe your feature request related problem
`AddWaitingOperator` is used to add operators.
image

From the code above, we can get that the rest operators in the loop will be ignored when `checkAddOperator` is false 
But currently, PD has supported adding batch operators such as in `evict_leader`. It maybe impacts the speed of scheduler.

### Describe the feature you'd like

Only ignore the false operators

### Describe alternatives you've considered

Continue the execution of the loop

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: Close #4643

### What is changed and how it works?
 Skip failed operator and continue executing the loop. And finally, execute `PromoteWaitingOperator` as many times as the length of ops.

```commit-message
Skip failed operator to support batch operators
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/schedule/operator_controller.go [hunk 1]
```diff
@@ -378,29 +383,29 @@ func (oc *OperatorController) checkAddOperator(ops ...*operator.Operator) bool {
 	for _, op := range ops {
 		region := oc.cluster.GetRegion(op.RegionID())
 		if region == nil {
-			log.Debug("region not found, cancel add operator",
```
Comment (rleungx) [inline server/schedule/operator_controller.go:381]: Please change it back to debug level.

Comment (CabinfeverB) [reply server/schedule/operator_controller.go]: fix

File: server/schedule/operator_controller.go [hunk 2]
```diff
@@ -306,11 +307,15 @@ func (oc *OperatorController) AddWaitingOperator(ops ...*operator.Operator) int
 		operatorWaitCounter.WithLabelValues(desc, "put").Inc()
 		oc.wopStatus.ops[desc]++
 		added++
+		needPromoted++
```
Comment (bufferflies) [inline server/schedule/operator_controller.go:310]: what's difference between added and needPromoted?

**Review by rleungx — APPROVED** [review]

File: server/schedule/operator_controller.go (hunk 2 above)
Comment (CabinfeverB) [reply server/schedule/operator_controller.go:310]: `needPromoted` is used for the promoting batch. if `isMerge` is true,  `added` will add 2 and `needPromoted` will add 1

File: server/schedule/operator_controller.go [hunk 3]
```diff
@@ -306,11 +307,15 @@ func (oc *OperatorController) AddWaitingOperator(ops ...*operator.Operator) int
...
 	}
 
 	oc.Unlock()
-	operatorWaitCounter.WithLabelValues(ops[0].Desc(), "promote-add").Inc()
-	oc.PromoteWaitingOperator()
+	for i := 0; i < needPromoted; i++ {
+		// these ops are created by one kind of scheduler/checker, so descs should be same
+		operatorWaitCounter.WithLabelValues(ops[0].Desc(), "promote-add").Inc()
```
Comment (bufferflies) [inline server/schedule/operator_controller.go:316]: Using  `add(float)` maybe better.

Comment (CabinfeverB) [reply server/schedule/operator_controller.go]: > Using `add(float)` maybe better.

fix

**Review by bufferflies — APPROVED** [review]

**Pull request merged**